src/RecordSession.cc

```diff
@@ -171,7 +171,11 @@ static bool handle_ptrace_exit_event(RecordTask* t) {
       // has been updated to reflect syscall entry. If we record a SCHED in
       // that state replay of the SCHED will fail. So detect that state and fix
       // it up.
-      if (looks_like_syscall_entry(t)) {
+      // If we got killed in an untraced syscall,
+      // there's no way on aarch64 that we can distinguish between
```

Comment:
There's PTRACE_GET_SYSCALL_INFO. Also that is technically what x7 tells you on aarch64, but that's terrible of course.

Reply:
I did see that after I wrote this fix. I wasn’t sure if that’s usable here both because we are not at a syscall stop and the tracee might be dead already…. The manual for that says “Retrieve information about the system call that caused the stop.” But we got here due to an exit…

it seems to me that for untraced syscall not recording them shouldn’t matter here since it should be indistinguishable from the case where we got killed just before or after syscall (I.e. the period between starting the buffered syscall till committing it should have no side effects that needs to be replayed on exit.) I assume the case this is handling is for an unbuffered syscall.